testsuite/gnat2goto/tests/recursion/recursion.adb

```diff
@@ -3,6 +3,6 @@ begin
    if (A <= 0) then
       return;
    else
-      Recursion (A - 1);
+      Recursion (A + 1);
```

Comment:
? Are we testing for non-termination?

Reply:
I've noticed in (some) other tests there is something along the lines of 
--  CBMC SUCCESS 
prior to the check. Should we do that here?

Reply:
> ? Are we testing for non-termination?

No, I had to change it to something that's already being handled in `gnat2goto`, because it doesn't handle operator minus.

> I've noticed in (some) other tests there is something along the lines of
> -- CBMC SUCCESS
> prior to the check. Should we do that here?

No, that's just an ADA comment that was documenting my expectations and I figured I should leave it there because it guides the expectations of the reader and makes it easier to verify his intuition with actual results. I don't think that's needed here if I'm honest.

Reply:
@sonodtt To clarify, the test is right now just testing that we can compile a recursive procedure and have cbmc correctly deal with the generated symbol table.

In this case it turns out the problem wasn't the recursion at all, we were just doing procedure calls wrong in general.